### PR TITLE
wip Label fix

### PR DESCRIPTION
**Pull request by posix4e**

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
Could you just push the Prometheus and rustfmt changes from here? That will leave the actual labels changes here. By themselves and we can sort out those things by themselves.

**Comment by posix4e** [comment]
@waywardmonkeys OK just pushed those two. Any better?

**Comment by waywardmonkeys** [comment]
I haven't forgotten about this. I am away from home this week but will comment on this in about 10-12 hours.

File: src/reporter/console.rs [hunk 1]
```diff
@@ -87,10 +87,11 @@ mod test {
...
         let mut r = StdRegistry::new();
-        r.insert("meter1", Metric::Meter(m.clone()));
-        r.insert("counter1", Metric::Counter(c.clone()));
-        r.insert("gauge1", Metric::Gauge(g.clone()));
-        r.insert("histogram", Metric::Histogram(h));
+        let labels = Vec::new();
+        r.insert("meter1", Metric::Meter(m.clone()), labels.clone());
+        r.insert("counter1", Metric::Counter(c.clone()), labels.clone());
+        r.insert("gauge1", Metric::Gauge(g.clone()), labels.clone());
+        r.insert("histogram", Metric::Histogram(h), labels);
```
Comment (waywardmonkeys) [inline src/reporter/console.rs]: Just use `vec![]` at each site instead of creating a vec and then cloning it?

File: src/registry.rs [hunk 2]
```diff
@@ -131,6 +117,6 @@ mod test {
             .build()
             .unwrap();
         h.increment_by(1, 1).unwrap();
-        r.insert("histogram", Metric::Histogram(h));
+        r.insert("histogram", Metric::Histogram(h), Vec::new());
```
Comment (waywardmonkeys) [inline src/registry.rs]: Use `vec![]` for all of these `Vec::new()`

File: src/reporter/prometheus.rs [hunk 3]
```diff
@@ -204,10 +204,13 @@ mod test {
         h.increment_by(1, 1).unwrap();
 
         let mut r = StdRegistry::new_with_labels(HashMap::new());
-        r.insert("meter1", Metric::Meter(m.clone()));
-        r.insert("counter1", Metric::Counter(c.clone()));
-        r.insert("gauge1", Metric::Gauge(g.clone()));
-        r.insert("histogram", Metric::Histogram(h));
+        let labels = Vec::new();
+        labels.push(("test", "bar"));
```
Comment (waywardmonkeys) [inline src/reporter/prometheus.rs]: `let labels = vec![("test", "bar")];` ?

Comment (posix4e) [inline src/reporter/prometheus.rs]: THis branch is going away,  it was just more of a test

**Pull request closed without merging**